ui/pages/permissions-connect/snap-install/snap-install.js

```diff
@@ -0,0 +1,112 @@
+import PropTypes from 'prop-types';
+import React, { useCallback, useMemo } from 'react';
+import { useDispatch } from 'react-redux';
+import { PageContainerFooter } from '../../../components/ui/page-container';
+import PermissionsConnectPermissionList from '../../../components/app/permissions-connect-permission-list';
+import PermissionsConnectFooter from '../../../components/app/permissions-connect-footer';
+import PermissionConnectHeader from '../../../components/app/permissions-connect-header';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import { hideModal, showSnapInstallWarning } from '../../../store/actions';
+
+export default function SnapInstall({
+  request,
+  approveSnapInstall,
+  rejectSnapInstall,
+  targetSubjectMetadata,
+}) {
+  const t = useI18nContext();
+  const dispatch = useDispatch();
+
+  const onCancel = useCallback(() => rejectSnapInstall(request.metadata.id), [
+    request,
+    rejectSnapInstall,
+  ]);
+  const onSubmit = useCallback(() => approveSnapInstall(request), [
+    request,
+    approveSnapInstall,
+  ]);
+  const showWarning = useCallback(
+    () =>
+      dispatch(
+        showSnapInstallWarning(targetSubjectMetadata.name, () => {
+          dispatch(hideModal());
+          approveSnapInstall(request);
+        }),
+      ),
+    [request, targetSubjectMetadata.name, approveSnapInstall, dispatch],
+  );
+
+  const npmId = useMemo(() => {
+    if (!targetSubjectMetadata.origin.startsWith('npm:')) {
+      return undefined;
+    }
+    return targetSubjectMetadata.origin.substring(4);
+  }, [targetSubjectMetadata]);
+  const shouldShowWarning = useMemo(
+    () =>
+      Boolean(
+        Object.keys(request.permissions).find((v) =>
+          v.startsWith('snap_getBip44Entropy_'),
+        ),
+      ),
+    [request.permissions],
+  );
+
+  return (
+    <div className="page-container snap-install">
+      <div className="headers">
+        <PermissionConnectHeader
+          icon={targetSubjectMetadata.iconUrl}
+          iconName={targetSubjectMetadata.name}
+          headerTitle={t('snapInstall')}
+          headerText={null} // TODO(ritave): Add header text when snaps support description
+          siteOrigin={targetSubjectMetadata.name}
+          npmPackageName={npmId}
+        />
+        <div className="snap-requests-permission">
+          {t('snapRequestsPermission')}
+        </div>
+        <PermissionsConnectPermissionList permissions={request.permissions} />
+      </div>
+      <div className="footers">
```

Comment:
Nit: The format we use for CSS classes is BEM - Block, Element, Modifier. See here for details: http://getbem.com/introduction/

The basic idea is that any component that makes sense on its own, as an independent component, is a "block", and each sub-piece that is a part of a greater whole is an element. And modifiers are for temporary states like "disabled", often not used.

So for the class names in this component to be BEM-compliant, this would be something like `snap-install__footer`.

Not a blocker, these class names work as-is functionally. Just wanted to make a note for future reference at least though, so your class names can be in-line with those used elsewhere in the wallet.